### DISCUSSION TITLE
Duppa.net 4 TTL Input USB Adapter

### DISCUSSION THREAD

**Discussion by claib** [Louisvdw/dbus-serialbattery#308] [Show and tell]
Hi there 
great success by integrating this fantastic ISOLATED ! 4 Input FTDI Adapter with ONE USB Port into Serialbattery.
Each FTDI port is isolated, so i had to power the Input with the help of a  12v > 3.5v step-down converter.
The 12v positive and ground i took directly from the " GPS " port of my JK-BMS.
The dupa.net "debug board quad " is showing up within Venus as .ttyUSB0 up to .ttyUSB3.
Works like a charm, an i have the other 3 FTDI Ports left for my other Victron VE-direct devices.
You can order a version with the VE-direct cables included !
https://www.duppa.net/shop/isolated-4-port-usb-to-ttl-uart-ftdi/
I think due to great success, they are sold out, but preorders possible.
Beside the necessity to power the TTL ( input side ) with the correct 3.3 v level, i was facing troubles to power my Raspi correctly, because there is a certain power consumption accumulating ;-)

**Comment by mr-manuel** [comment]
Hi @claib, would it be possible that you post more informations about this adapter? Maybe this is a possible solution for changing DeviceId (https://github.com/Louisvdw/dbus-serialbattery/discussions/481).

Could you also post the output of `lsusb`?

**Reply by claib** [reply]
hi there
at the moment i'm afraid that i can's give you a lsusb output, because my installation is far....
You can order the adapter with the provided link. There a single, and 4 port devices with isolation and victron cables attached.

**Reply by mr-manuel** [reply]
You don't have SSH access? Please set a reminder for the next time you pass, or maybe someone else has already such an adapter? Off Grid Garage has one, but I don't know his username on GitHub.

**Reply by Sarah-1331** [reply]
i have one how can i help? currently installed

**Reply by mr-manuel** [reply]
Connect via SSH to the GX device and enter `lsusb`. Post the result of the command.

**Reply by Sarah-1331** [reply]
```bash
Last login: Sun Nov 26 00:53:36 2023
root@raspberrypi2:~# 
root@raspberrypi2:~# 
root@raspberrypi2:~# lsusb
Bus 001 Device 005: ID 0403:6011 Future Technology Devices International, Ltd FT4232H Quad HS USB-UART/FIFO IC

Bus 001 Device 004: ID 0403:6015 Future Technology Devices International, Ltd Bridge(I2C/SPI/UART/FIFO)

Bus 001 Device 006: ID 0424:7800 Standard Microsystems Corp. 

Bus 001 Device 003: ID 0424:2514 Standard Microsystems Corp. USB 2.0 Hub

Bus 001 Device 002: ID 0424:2514 Standard Microsystems Corp. USB 2.0 Hub

Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
root@raspberrypi2:~#
```

**Reply by Sarah-1331** [reply]
I can confirm my mppt chargers don’t change device is in VRM anymore either since fitting
